src/test/java/com/example/demo/DemoApplicationTests.java

```diff
@@ -1,19 +1,19 @@
 package com.example.demo;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
 import com.example.demo.helpers.HoverflyHelper;
 import io.specto.hoverfly.junit.core.Hoverfly;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import reactor.netty.http.client.HttpClient;
-
-import java.io.IOException;
-
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.test.context.ActiveProfiles;
 
-@SpringBootTest
+@ActiveProfiles("test")
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
```

Comment:
need to spin up the servelet for making real HTTP call to your `helloworld` endpoint on port 8080